### PR TITLE
Gather should not be considered a gradient operation.

### PR DESCRIPTION
**Pull request by tex3d**
Also fixed:
- remove SampleCmpLevelZero in DxilConvergent
- next_op_idx assert message in hctdb.py
- add WriteSamplerFeedback to gradient and convergent ops

#2759 is fixed by this, but will require updated validator.

### CODE REVIEW

File: utils/hct/hctdb.py [hunk 1]
```diff
@@ -1749,11 +1749,11 @@ def populate_dxil_operations(self):
 
         # End of DXIL 1.6 opcodes.
         self.set_op_count_for_version(1, 6, next_op_idx)
-        assert next_op_idx == 218, "next operation index is %d rather than 165 and thus opcodes are broken" % next_op_idx
+        assert next_op_idx == 218, "218 is expected next operation index but encountered %d and thus opcodes are broken" % next_op_idx
```
Comment (pow2clk) [inline utils/hct/hctdb.py:1752]: Hah! I was going to fix this in an unrelated change. This wording is less confusing.

**Comment by pow2clk** [comment]
The build break is pre-existing. I say submit or redo the build at your discretion.

File: tools/clang/test/HLSLFileCheck/hlsl/objects/Texture/cube_gather.hlsl [hunk 2]
```diff
@@ -18,64 +18,65 @@ float4 main(float4 a : A) : SV_Target
   ///////////////////////////////////////////////
 
   // CHECK: call %dx.types.ResRet.f32 @dx.op.textureGather.f32(i32 73,
-  // CHECK: , float %{{[^,]+}}, float %{{[^,]+}}, float %{{[^,]+}}, float undef
-  // CHECK: , i32 undef, i32 undef
-  // CHECK: , i32 0)
+  // CHECK-SAME: , float %{{[^,]+}}, float %{{[^,]+}}, float %{{[^,]+}}, float undef
```
Comment (tex3d) [inline tools/clang/test/HLSLFileCheck/hlsl/objects/Texture/cube_gather.hlsl:21]: Changed all these to CHECK-SAME because debugging the test when it failed was much harder without it.
